docs/structured-streaming-programming-guide.md

```diff
@@ -0,0 +1,888 @@
+is not currently receiving any data as we are just setting up the 
+transformation, and have not yet started it. Next, we have used to built-in 
+SQL functions - split and explode, to split each line into multiple rows with 
+a word each. In addition, we use the function `alias` to name the new column 
+as “word”. Finally, we have defined the running counts, by grouping the `words`
+DataFrame by the column `word` and count on that grouping. 
+
+We have now set up the query on the streaming data. All that is left is to 
+actually start receiving data and computing the counts. To do this, we set it 
+up to output the counts to the console every time they are updated. In 
+addition we are also going to set up additional details like checkpoint 
+location. Don’t worry about them for now, they are explained later in the guide. 
+
+<div class="codetabs">
+<div data-lang="scala"  markdown="1">
+
+{% highlight java %}
+val query = wordCounts
+  .writeStream
+  .outputMode("complete")
+  .format("console")
+  .option("checkpointLocation", checkpointDir)
+  .start()
+
+query.awaitTermination()
+{% endhighlight %}
+
+</div>
+<div data-lang="java"  markdown="1">
+
+{% highlight java %}
+StreamingQuery query = wordCounts
+    .writeStream()
+    .outputMode("complete")
+    .format("console")
+    .option("checkpointLocation", checkpointDir)
+    .start();
+
+query.awaitTermination();
+{% endhighlight %}
+
+</div>
+<div data-lang="python"  markdown="1">
+
+{% highlight python %}
+query = wordCounts\
+    .writeStream\
+    .outputMode('complete')\
+    .format('console')\
+    .option('checkpointLocation', checkpointDir)\
+    .start()
+
+query.awaitTermination()
+{% endhighlight %}
+
+</div>
+</div>
+
+Now the streaming computation has started in the background, and the `query` object is a handle to that active streaming query. Note that we are also waiting for the query to terminate, to prevent the process from finishing while the query is active.
+To actually run this code, you can either compile your own Spark application, or simply run the example once you have downloaded Spark. We are showing the latter. You will first need to run Netcat (a small utility found in most Unix-like systems) as a data server by using
+
+    $ nc -lk 9999
+
+Then, in a different terminal, you can start the example by using
+
+<div class="codetabs">
+<div data-lang="scala"  markdown="1">
+
+    $ ./bin/run-example org.apache.spark.examples.sql.streaming.StructuredNetworkWordCount
+
+</div>
+<div data-lang="java"  markdown="1">
+
+    $ ./bin/run-example org.apache.spark.examples.sql.streaming.JavaStructuredNetworkWordCount 
+
+</div>
+<div data-lang="python"  markdown="1">
+    
+    $ ./bin/spark-submit examples/src/main/python/sql/streaming/structured_network_wordcount.py
+
+</div>
+</div>
+
+Then, any lines typed in the terminal running the netcat server will be counted and printed on screen every second. It will look something like the following.
```

Comment:
I don't see an example here... you might just be able to remove that last sentence?